colossalai/booster/plugin/moe_hybrid_parallel_plugin.py

```diff
@@ -0,0 +1,339 @@
+import random
+from typing import Any, Callable, Iterator, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.distributed as dist
+from torch.nn import Module
+from torch.optim import Optimizer
+from torch.optim.lr_scheduler import _LRScheduler as LRScheduler
+from torch.utils.data import DataLoader
+from torch.utils.data.distributed import DistributedSampler
+
+from colossalai.booster.plugin.hybrid_parallel_plugin import (
+    HybridParallelAMPOptimizer,
+    HybridParallelModule,
+    HybridParallelNaiveOptimizer,
+    HybridParallelZeroOptimizer,
+    get_param_info,
+)
+from colossalai.checkpoint_io import CheckpointIO, HybridParallelCheckpointIO
+from colossalai.cluster import ProcessGroupMesh
+from colossalai.interface import ModelWrapper, OptimizerWrapper
+from colossalai.pipeline.schedule import OneForwardOneBackwardSchedule
+from colossalai.pipeline.stage_manager import PipelineStageManager
+from colossalai.shardformer import ShardConfig
+from colossalai.shardformer.policies.base_policy import Policy
+
+from .pp_plugin_base import PipelinePluginBase
+
+PP_AXIS, DP_AXIS, TP_AXIS = 0, 1, 2
+
+
+class MoeHybridParallelPlugin(PipelinePluginBase):
```

Comment:
What's the difference between this plugin and hybrid parallel plugin?

Reply:
their process grouph axis is different

Reply:
Then can it inherit HybridParallelPlugin and overwrite a few methods like `__init__()`?

Reply:
ok